### PR TITLE
docs: Move table titles from CompleteXMLSchema.rst to corresponding include file

### LINKED ISSUE #3316: Fixing titles in searchable docs
Right now, of one searches a term in the docs the sub-tables also appear as separate entries - see below when I search for SourceFlux. The titles should just be moved into the individual files.

Screen Shot 2024-08-29 at 11 19 20 AM

### PR DESCRIPTION
**Pull request by ryar9534**
Fixes #3316 by moving the title of each table into the corresponding file.

Before this pr, if you search for cflFactor (for example) in the docs you will see
image

With this PR the results become
image

### CODE REVIEW

**Comment by ryar9534** [comment]
I am somewhat tempted to change 'Element' to 'XML Element' or something, so that when one searches it is clear what the element is. Thoughts @rrsettgast ?

**Comment by rrsettgast** [comment]
> I am somewhat tempted to change 'Element' to 'XML Element' or something, so that when one searches it is clear what the element is. Thoughts @rrsettgast ?

That seems fine to me....

one thing that I don't understand is that these rst files are all automatically generated, but I don't see any changes to the code/script that generates them?? Am I missing something?

**Comment by ryar9534** [comment]
> > I am somewhat tempted to change 'Element' to 'XML Element' or something, so that when one searches it is clear what the element is. Thoughts @rrsettgast ?
> 
> That seems fine to me....
> 
> one thing that I don't understand is that these rst files are all automatically generated, but I don't see any changes to the code/script that generates them?? Am I missing something?

Ah yes I forgot to specify - I just changed them manually to make sure I understood what was happening, I was planning on now digging deeper into the scripts that generate them

File: src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp [hunk 1]
```diff
@@ -35,9 +35,9 @@ template< std::ptrdiff_t N >
 class LinearSystem
 {
 public:
-  real64 matrix[N][N];
-  real64 solution[N];
-  real64 rhs[N];
+  real64 matrix[N][N] = {0};
+  real64 solution[N] = {0};
+  real64 rhs[N] = {0};
```
Comment (rrsettgast) [inline src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp]:
```suggestion
  real64 matrix[N][N] = {};
  real64 solution[N] = {};
  real64 rhs[N] = {};
```

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp (hunk 1 above)
Comment (CusiniM) [reply src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp]: why are these changes in this PR?

File: src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp [hunk 2]
```diff
@@ -35,9 +35,9 @@ template< std::ptrdiff_t N >
 class LinearSystem
 {
 public:
-  real64 matrix[N][N];
-  real64 solution[N];
-  real64 rhs[N];
+  real64 matrix[N][N] = {};
+  real64 solution[N] = {};
+  real64 rhs[N] = {};
```
Comment (CusiniM) [inline src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp]:
```suggestion
  real64 matrix[N][N]{};
  real64 solution[N]{};
  real64 rhs[N] {};
```

it's equivalent but this is how it's done in the rest of the code

**Review by paveltomin — APPROVED** [review]

**Comment by ryar9534** [comment]
@rrsettgast Im not sure I understand why the readthedocs build is failing. Looks like this is the error in the log:

```
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/events/docs/EventManager.rst:58: CRITICAL: Title level inconsistent:

PeriodicEvent
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/events/docs/EventManager.rst:73: CRITICAL: Title level inconsistent:

SoloEvent
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/events/docs/EventManager.rst:80: CRITICAL: Title level inconsistent:

HaltEvent
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/events/docs/TasksManager.rst:26: CRITICAL: Title level inconsistent:

PackCollection
***************************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/events/docs/TasksManager.rst:36: CRITICAL: Title level inconsistent:

***************************
Triggering the Tasks
***************************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:70: CRITICAL: Title level inconsistent:

1D Table
*************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:85: CRITICAL: Title level inconsistent:

ND Table
*************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:110: CRITICAL: Title level inconsistent:

Interpolation Methods
************************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:129: CRITICAL: Title level inconsistent:

Table Generation Example
**************************
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:154: CRITICAL: Title level inconsistent:

SymbolicFunction
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/geosx-geosx/checkouts/3321/src/coreComponents/functions/docs/FunctionManager.rst:184: CRITICAL: Title level inconsistent:

CompositeFunction
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by rrsettgast** [comment]
@ryar9534 
This is because the header levels are inconsistent. Here are the headers in RST:
```
# with overline, for parts
* with overline, for chapters
= for sections
- for subsections
^ for subsubsections
" for paragraphs
```
This is taken from:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

So now you have put the title headers (which are `======`) in the table RST files, and when referenced from `CompleteXMLSchema.rst` they are at the correct level as they fall under `******`. However when these files are referenced elsewhere in the code, as is the case from each of these warnings/errors, they full under different header levels. In the example I looked at, this is `^^^^^^^`, so this is incorrect. We can try to use a lower level header on the table, but that may still give an warning/error.

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**